tests/unit/src/test/scala/tests/DebugDiscoverySuite.scala

```diff
@@ -0,0 +1,253 @@
+package tests
+
+import java.util.concurrent.TimeUnit
+
+import scala.meta.internal.metals.DebugDiscoveryParams
+import scala.meta.internal.metals.JsonParser._
+import scala.meta.internal.metals.debug.BuildTargetContainsNoMainException
+import scala.meta.internal.metals.debug.NoTestsFoundException
+import scala.meta.internal.metals.debug.WorkspaceErrorsException
+
+// note(@tgodzik) all test have `System.exit(0)` added to avoid occasional issue due to:
+// https://stackoverflow.com/questions/2225737/error-jdwp-unable-to-get-jni-1-2-environment
+class DebugDiscoverySuite extends BaseDapSuite("debug-discovery") {
+  private val mainPath = "a/src/main/scala/a/Main.scala"
+  private val fooPath = "a/src/main/scala/a/Foo.scala"
+  private val barPath = "a/src/main/scala/a/Bar.scala"
+
+  test("run".tag(FlakyWindows)) {
```

Comment:
What is the issue on windows? Most other tests work fine there and we haven't had the need to ignore much on Windows. It seems kind of harsh to make the entire suite flaky. We will never seen any issues pop up there.

Reply:
So there isn't any other test suites that actually "pretend" to send in a URI the same way that a client would send it in. The trick part here is that the `${workspace}` is always nice and done in the OS specific way. However, the path that we put together, like in this example, `mainPath`, is done in unix fashion with `a/src/main/scala/a/Main.scala`. Then the testing server takes that apart and makes sure that it is created correctly for the OS. However, how do you then get that exact created path back to pretend to pass it into the server for it to match that it's indeed there. Everything I tried resulted in a screwed up `file:///D:\a\metals\metals\tests\unit\target\e2e\debug-discovery\run/a/src/main/scala/a/Main.scala` like syntax. I even tried a gross `replaceAll` on the `mainPath` variable when `isWindows` was detected. At that point it was getting pretty hacky and I just ignored windows. Any idea how to accurately get this file URI in the OS specific way in the test to match exactly how a client would send it? I just pushed this up with the `flakyWindows` removed from the `run` test just to show you.